src/components/opt-in-out/index.tsx

```diff
@@ -92,13 +98,19 @@ export default function OptInOut({ platform, redirectPath }: OptInOutProps) {
         iconPosition="trailing"
         onClick={openDialog}
       />
-      <Dialog onDismiss={closeDialog} isOpen={showDialog} label="Opt out form">
-        <OptOutForm
-          onSubmit={handleOptOut}
+      {isDialogPreloaded || showDialog ? (
+        <Dialog
           onDismiss={closeDialog}
-          platform="learn"
-        />
-      </Dialog>
+          isOpen={showDialog}
+          label="Opt out form"
+        >
+          <OptOutForm
+            onSubmit={handleOptOut}
+            onDismiss={closeDialog}
+            platform="learn"
+          />
+        </Dialog>
+      ) : null}
```

Comment:
This is behind the conditional because if you just render a `next/dynamic` component, it will load the assets right away. In this case, we don't want that to happen, so we add the `isDialogPreloaded` check and only render the component if someone clicks the button OR the component is loaded.

Reply:
Does this impact any animation/transitions on `Dialog`?

Reply:
Nope, that's why `showDialog` is part of the condition. If someone tries to open it, it'll render as normal so the animations will work.